rllib/offline/offline_prelearner.py

```diff
@@ -526,20 +508,64 @@ def _map_sample_batch_to_episode(
                 NotImplementedError
             else:
                 # Unpack observations, if needed.
-                obs = (
-                    unpack_if_needed(obs.tolist())
-                    if schema[Columns.OBS] in input_compress_columns
-                    else obs.tolist()
-                )
-                # Append the last `new_obs` to get the correct length of observations.
-                obs.append(
-                    unpack_if_needed(batch[schema[Columns.NEXT_OBS]][i][-1])
-                    if schema[Columns.OBS] in input_compress_columns
-                    else batch[schema[Columns.NEXT_OBS]][i][-1]
-                )
+                if isinstance(obs, str):
+                    # Decompress the observations
+                    obs = unpack_if_needed(obs)
+                    # Convert to a list of arrays.
+                    obs = [obs[i, ...] for i in range(obs.shape[0])]
+                else:
+                    obs = (
+                        unpack_if_needed(obs.tolist())
+                        if schema[Columns.OBS] in input_compress_columns
+                        else obs.tolist()
+                    )
+
+                if schema[Columns.NEXT_OBS] in batch:
+                    # Append the last `new_obs` to get the correct length of
+                    # observations.
+                    obs.append(
+                        unpack_if_needed(batch[schema[Columns.NEXT_OBS]][i][-1])
+                        if schema[Columns.OBS] in input_compress_columns
+                        else batch[schema[Columns.NEXT_OBS]][i][-1]
+                    )
+                else:
+                    # Otherwise we duplicate the last observation.
+                    obs.append(obs[-1])
+
+                # Check, if we have `done`, `truncated`, or `terminated`s in
+                # the batch.
+                if (
+                    schema[Columns.TRUNCATEDS] in batch
+                    and schema[Columns.TERMINATEDS] in batch
+                ):
+                    truncateds = batch[schema[Columns.TRUNCATEDS]][i][-1]
```

Comment:
super nit: Rename "truncateds" into "truncated" and "terminateds" into "terminated".

We should only use the plural if these are a sequence, not for individual bools.